integration-test/1810-alternate-viewpoints.py

```diff
@@ -394,3 +393,147 @@ def test_none_none(self):
+                'border_type': 'province',
+                'boundary': 'administrative',
+                'is_in:country_code': 'XA',
+                'name': 'XA region 1',
+                'type': 'boundary',
+                'mz_boundary_from_polygon': True,  # need this for hack
+            }),
+            # this one we do pay attention to, admin_level=4
+            dsl.way(3, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '4',
+                'boundary': 'administrative',
+                'name': 'XA region 2',
+                'place': 'District',
+                'type': 'boundary',
+                'mz_boundary_from_polygon': True,  # need this for hack
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'region',
+                'kind:xb': 'unrecognized_region',
+                'name': 'XA region 2',
+            })
+
+    def test_claim(self):
+        import dsl
+
+        z, x, y = 16, 0, 0
+
+        self.generate_fixtures(
+            # XA's claim relation
+            dsl.way(1, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '4',
+                'boundary': 'claim',
+                'name': 'XA region claim',
+                'claimed_by': 'XA',
+                'disputed_by': 'XB',
+                'source': 'openstreetmap.org',
+            }),
+            # XA's claim _way_, disputed by XB
+            dsl.way(2, dsl.tile_diagonal(z, x, y), {
+                'disputed': 'yes',
+                'disputed_by': 'XB',
+                'source': 'openstreetmap.org',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'unrecognized_region',
+                'kind:xa': 'region',
+            })
+
+
+class CountyBoundary(FixtureTest):
+
+    def test_dispute(self):
+        import dsl
+
+        z, x, y = 16, 0, 0
+
+        self.generate_fixtures(
+            # way, tagged disputed
+            dsl.way(1, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '4',
+                'boundary': 'administrative',
+                'disputed': 'yes',
+                'disputed_by': 'XB',
+                'name': 'XA internal county border',
+                'source': 'openstreetmap.org',
+            }),
+            # line from relation / county polygon
+            dsl.way(3, dsl.tile_diagonal(z, x, y), {
+                'admin_level': '6',
+                'boundary': 'administrative',
+                'name': 'XA county 2',
+                'place': 'District',
+                'type': 'boundary',
+                'mz_boundary_from_polygon': True,  # need this for hack
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'boundaries', {
+                'kind': 'county',
+                'kind:xb': 'unrecognized_county',
+                'name': 'XA county 2',
```

Comment:
Because of the different admin levels here, it looks like it takes the disputed from the `county` level (4), but the name from the localadmin level (6). Should it be the level 4 name instead?

Reply:
`admin_level=6` is for counties. We use `admin_level=4` for regions.